src/Core/CommandBus/Parser/CommandHandlerDefinitionParser.php

```diff
@@ -0,0 +1,204 @@
+    {
+        $method = $handlerReflection->getMethod(self::HANDLER_METHOD_NAME);
+
+        foreach ($handlerReflection->getInterfaces() as $interface) {
+            if ($interface->hasMethod(self::HANDLER_METHOD_NAME)) {
+                $method = $interface->getMethod(self::HANDLER_METHOD_NAME);
+                break;
+            }
+        }
+
+        if ($returnType = $this->parseReturnTypeFromDocblock($method)) {
+            return $returnType;
+        }
+
+        if ($method->hasReturnType()) {
+            /* @phpstan-ignore-next-line ignoring getName() as method is not visible to phpstan */
+            return $method->getReturnType()->getName();
+        }
+
+        return null;
+    }
+
+    /**
+     * @param ReflectionMethod $method
+     *
+     * @return string|null
+     */
+    private function parseReturnTypeFromDocblock(ReflectionMethod $method): ?string
+    {
+        $docBlock = $method->getDocComment();
+        if (!$docBlock) {
+            return null;
+        }
+
+        $tagPosition = strpos($docBlock, self::RETURN_TAG);
+
+        if (false !== $tagPosition) {
+            $returnType = substr($docBlock, $tagPosition, (strpos($docBlock, PHP_EOL, $tagPosition)) - $tagPosition);
+            $returnType = str_replace(sprintf('%s ', self::RETURN_TAG), '', $returnType);
+
+            return $returnType;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param ReflectionClass $reflectionClass
+     *
+     * @return string
+     */
+    private function parseDescription(ReflectionClass $reflectionClass): string
+    {
+        if (!$docBlock = $reflectionClass->getDocComment()) {
+            return '';
+        }
+
+        /**
+         * Removes comment symbols, annotations, and line breaks.
+         */
+        $description = preg_replace("/\/+\*\*|\*+\/|\*|@(\w+)\b(.*)|\n/",
+            '',
+            $docBlock
+        );
+
+        /**
+         * Replaces multiple spaces to single space
+         */
+        $description = preg_replace('/ +/', ' ', $description);
+
+        /*
+         * Strips whitespace from the beginning and end
+         */
+        return trim($description);
+    }
+
+    /**
+     * Checks whether the command is of type Query or Command by provided name
+     *
+     * @param string $commandName
+     *
+     * @return string command|query
+     */
+    private function parseType(string $commandName): string
+    {
+        if (strpos($commandName, '\Command\\')) {
+            return CommandHandlerDefinition::TYPE_COMMAND;
+        }
+
+        return CommandHandlerDefinition::TYPE_QUERY;
+    }
+
+    /**
+     * @param string $commandClass
+     *
+     * @return string
+     */
+    private function parseDomain(string $commandClass): string
+    {
+        preg_match('/PrestaShop\\\\PrestaShop\\\\Core\\\\Domain\\\\([a-zA-Z]+)\\\\/', $commandClass, $matches);
+
+        return $matches[1];
```

Comment:
So you assume it always match something?

Reply:
In our limited use case yes, please keep in mind this is a tool that help us build some doc automatically
So it doesn't need to be extra careful as the input is scoped, its purpose is not to be used in any project but for a very specific use case, it can always be improved if someday we find some bugs while we need to update the doc

Reply:
The code shouldn't be considered as a tool that can be improved, commands are in the release package and can be used by anyone. Otherwise, if it's only for ourselves, it should be in a dedicated repository :thinking: